Implement LookupAsn as a replacement of getasn in cnc.go

coolparadox opened the pull request:
@sajal LookupAsn does not implement cache nor queries to our own database yet (these will be further issues).

Fixes #1

coolparadox committed:
CymruDnsLookup stub

References #1

coolparadox committed:
CymruDnsLookup implementation

The geoipdb handler now stores an initialized "cymruClient" object faster processing.
Also, thread safety is intended by mutex controlled access to cymruClient data.

References #1

coolparadox committed:
LookupAsn uses team cymru service for finding ASN description.

This happens when other queries (libgeoip, ipinfo) returned empty
descriptions.

References #1

sajal commented:
https://github.com/turbobytes/geoipdb/pull/6/files#diff-7563c91ce4ebe025939e52d29001b48eR225

I see you are using locks there. We do not want to do that. This will make lookups sequential and slow. Imagine if each lookup takes 2 seconds, then we are stuck with doing 1 lookup per 2 seconds. A single test result might require 100s of lookups at a time. It is fine to create new `dns.Msg` for each lookup, and let garbage collector deal with it.

coolparadox committed:
mutex locks removed from geoipdb code.

Locks are evil here, as they serialize concurrent lookups. This heavily
impacts performance.

References #1
